worker/include/RTC/RTCP/ReceiverReport.hpp

```diff
@@ -190,12 +199,12 @@ namespace RTC
 			}
 			size_t GetSize() const override
 			{
-				size_t size = Packet::CommonHeaderSize + 4u /* this->ssrc */;
-
-				for (auto* report : reports)
-				{
-					size += report->GetSize();
-				}
+				// A serialized packet can contain a maximum of 31 reports.
+				// If number of reports exceeds 31 then the required number of packets
+				// will be serialized which will take the size calculated below.
+				size_t size = (Packet::CommonHeaderSize + 4u /* this->ssrc */) *
+				              ((this->GetCount() / MaxReportsPerPacket) + 1);
+				size += ReceiverReport::HeaderSize * this->GetCount();
```

Comment:
Can you explain in detail this math?

Reply:
A Receiver Report packet is composed by:

* RTCP common header
* SSRC of the packet sender.
* Up to 31 reports.

This method returns the size needed to represent all it's content.

Reply:
![image](https://user-images.githubusercontent.com/16191/197983587-d05c3826-1ffb-4441-94d4-92162781eb5c.jpeg)

Why is (commonHeader size + 4) being multiplied by ((number of reports divided by max reports per packet) + 1)?

Reply:
We are calculating there how many receiver report packets (Common header + SSRC) we need, considering the packet can hold more than 31 receiver reports, the maximum reports a serialized receiver report packet can contain (limit given by the 'count' header field in the RTCP common header)